### ISSUE TITLE
[Selfhost] Instance keeps redirecting to http

### ISSUE THREAD

**Issue by finzzz** [gristlabs/grist-core#187]
Currently, there is no way of setting https only with docker.
I needed to open port 80 and redirect it manually to https using reverse proxy.
I have tried setting APP_DOC_URL APP_HOME_URL GRIST_ROUTER_URL , etc.. but none works.

**Comment by paulfitz** [comment]
Not sure I entirely understand the question. The Grist docker image doesn't deal with https, that is true. There is a complete example of hosting Grist, with certificates using Let's Encrypt, at https://community.getgrist.com/t/a-template-for-self-hosting-grist-with-traefik-and-docker-compose/856

**Comment by finzzz** [comment]
What I need is for grist to also specify the protocol, instead of only domain.

**Comment by paulfitz** [comment]
Do you mean you'd like the Grist docker container to have something within it that:
 * terminates SSL connections (perhaps using certificates that you provide?)
 * handles redirection of http requests to https
so that you don't need to handle that yourself using a reverse proxy? If so, I can understand wanting that, and I'll mark this as a feature request. Just want to be sure you're not having some more fundamental problem with getting Grist working.

**Comment by finzzz** [comment]
Yes, it's sort of a feature request. Although I think it would be best to let reverse proxy handles ssl certs. I just need to be able to specify http scheme using variable.

**Comment by paulfitz** [comment]
> I just need to be able to specify http scheme using variable.

It really sounds like you are seeing a problem of some kind, but I can't guess what it is. Maybe #149 is relevant?

**Comment by finzzz** [comment]
Nope, I was able to get it working. But it would be cleaner if there is a way to specify protocol without having to open port 80 and redirect it to 443. The method you're suggesting using traefik will create more frictions as I have my own reverse proxy in place.

**Comment by paulfitz** [comment]
I don't understand what port 80 or 443 would have to do with the Grist container. If a user attempts to visit your site via http on port 80, it would be the job of the reverse proxy to redirect them, right? Grist should only see the traffic after ssl termination, passed along on port 8484 or whatever you choose?

**Comment by finzzz** [comment]
My ideal setup is to only open port 443 and map port 8484:8484 to grist container. Then the reverse proxy 443=> localhost:8484=>grist container. This setup is not possible, because grist container defaults the address to http. I needed to do reverse proxy 80=>redirect to 443=> localhost:8484 => grist container.

**Comment by paulfitz** [comment]
Can you tell me more about what you mean by `This setup is not possible, because grist container defaults the address to http`. What address?

**Comment by finzzz** [comment]
Grist container keeps redirecting me to http://gristdomain/o/docs instead of https://gristdomain/o/docs

**Comment by dsagal** [comment]
For your setup, I think the main thing you need is to set `APP_HOME_URL` to `https://gristdomain` (i.e. include the protocol and make it `https` -- it should be the URL for browsers to use). This is what's done in the link that @paulfitz shared earlier (https://community.getgrist.com/t/a-template-for-self-hosting-grist-with-traefik-and-docker-compose/856). That setup is similar to yours and is known to work.

BTW, I don't think either `APP_DOC_URL` or `GRIST_ROUTER_URL` is needed, but it's possible that setting `GRIST_SINGLE_ORG=grist` is important.

**Comment by finzzz** [comment]
Like I said many times, it doesn't address the issue that grist can't handle https. It creates more frictions by adding another layer of reverse proxy on top of my reverse proxy.

**Comment by Vladimir-Va** [comment]
Hello @finzzz i think you install docker without GRIST_SINGLE_ORG env.
This is a known issue that is currently being resolved, while it is not possible to install without
GRIST_SINGLE_ORG=myorganization
GRIST_ORG_IN_PATH=true
https://community.getgrist.com/t/grist-core-multi-user-docker-setup/666/8

**Comment by finzzz** [comment]
I tried everything after searching similar issues in the forum and github. None works.

**Comment by paulfitz** [comment]
Just to confirm, this is definitely a bug, and can be easily replicated. If you visit `https://gristdomain`, it will redirect to `http://gristdomain/o/docs`; it should instead redirect to `https://gristdomains/o/docs`. Installations that redirect from http to https hide the problem. This is without `GRIST_SINGLE_ORG` set. You can see the code building a redirect url incorrectly here: https://github.com/gristlabs/grist-core/blob/05c89fb65a3e377449e31ec1e3662c7254686b70/app/server/lib/FlexServer.ts#L1521
This doesn't work because of the reverse proxy, the request protocol will be `http` but the user should be directed to `https`.

**Comment by Vladimir-Va** [comment]
Just to clarify, now we can remove env:
GRIST_SINGLE_ORG=myorganization
And use GRIST_ORG_IN_PATH=true for multiple org for selfhosted install?

**Comment by paulfitz** [comment]
@Vladimir-Va if you wish to, yes - that is a separate issue though. The status of multi-org support hasn't changed recently (it works, but creating new orgs/teams is a manual process).
